cdap-ui/graphql/schemaQuery.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+// See ./graphql/fragements/README.md
+
+const fetch = require('node-fetch');
+const fs = require('fs');
+
+fetch(`http://localhost:11011/graphql`, {
+  method: 'POST',
+  headers: { 'Content-Type': 'application/json' },
+  body: JSON.stringify({
+    variables: {},
+    query: `
+      {
+        __schema {
+          types {
+            kind
+            name
+            possibleTypes {
+              name
+            }
+          }
+        }
+      }
+    `,
+  }),
+})
+  .then((result) => result.json())
+  .then((result) => {
+    // here we're filtering out any type information unrelated to unions or interfaces
+    const filteredData = result.data.__schema.types.filter((type) => type.possibleTypes !== null);
+    result.data.__schema.types = filteredData;
+    fs.writeFileSync(
+      './graphql/fragments/fragmentTypes.json',
+      JSON.stringify(result.data),
```

Comment:
nit: do `JSON.stringify(result.data).concat('\n')`

so that the fragment file doesn't have that EOF warning on GitHub

Reply:
Done